crates/biome_js_analyze/src/lint/nursery/no_useless_length_check.rs

```diff
@@ -0,0 +1,366 @@
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (array.length > 0 && array.some(Boolean));
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const isAllTrulyOrEmpty = array.length === 0 || array.every(Boolean);
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// if (array.every(Boolean));
+    /// ```
+    ///
+    /// ```js
+    /// if (array.some(Boolean));
+    /// ```
+    ///
+    /// ```js
+    /// const isAllTrulyOrEmpty = array.every(Boolean);
+    /// ```
+    ///
+    /// ```js
+    /// const isAllTrulyOrEmpty = array.every(Boolean);
+    /// ```
+    ///
+    /// ```js
+    /// if (array.length === 0 || anotherCheck() || array.every(Boolean));
+    /// ```
+    ///
+    /// ```js
+    /// const isNonEmptyAllTrulyArray = array.length > 0 && array.every(Boolean);
+    /// ```
+    ///
+    /// ```js
+    /// const isEmptyArrayOrAllTruly = array.length === 0 || array.some(Boolean);
+    /// ```
+    ///
+    pub NoUselessLengthCheck {
+        version: "next",
+        name: "noUselessLengthCheck",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintUnicorn("no-useless-length-check")],
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum ErrorType {
+    UselessLengthCheckWithSome,
+    UselessLengthCheckWithEvery,
+}
+
+/// Whether the node is a descendant of a logical expression.
+fn is_logical_exp_descendant(node: &AnyJsExpression, operator: JsSyntaxKind) -> bool {
+    let Some(parent) = node.syntax().parent() else {
+        return false;
+    };
+    parent
+        .ancestors()
+        .find_map(|ancestor| {
+            if let Some(logical_exp) = JsLogicalExpression::cast(ancestor.clone()) {
+                return logical_exp
+                    .operator_token()
+                    .ok()
+                    .map(|token| token.kind() == operator)
+                    .or(Some(false));
+            }
+            (!JsParenthesizedExpression::can_cast(ancestor.kind())).then_some(false)
+        })
+        .unwrap_or(false)
+}
+
+/// Extract the expressions that perform length comparisons corresponding to the errors you want to check.
+fn get_comparing_length_exp(
+    binary_exp: &JsBinaryExpression,
+    expect_error: &ErrorType,
+) -> Option<AnyJsExpression> {
+    let left = binary_exp.left().ok()?;
+    let operator = binary_exp.operator().ok()?;
+    let right = binary_exp.right().ok()?;
+
+    // Check only when the number appears on the right side according to the original rules.
+    // We assume that you have already complied with useExplicitLengthCheck
+    let compare_exp = left;
+    let value_exp = right;
+
+    let AnyJsExpression::JsStaticMemberExpression(member_exp) = compare_exp else {
+        return None;
+    };
+    let target = member_exp.object().ok()?;
+    let member = member_exp.member().ok()?;
+    if member.text() != "length" || member_exp.is_optional_chain() {
```

Comment:
```suggestion
    if member.text() != "length" || member_exp.is_optional_chain() {
```

As mentioned before, try not to use `.text` from a typed node, because it creates and allocates new `String`. Use `syntax().text_trimmed()`